File: Sheets/src/neetcode/practice/Math_And_Geometry/_6_powXN.java
package neetcode.practice.Math_And_Geometry;

public class _6_powXN {
  public static double myPow(double x, int n) {
    double res = 1.0;
    if (n > 0) {
      for (int i = 0; i < n; i++) res *= x;
    } else if (n < 0) {
      for (int i = 0; i < -n; i++) res *= 1.0 / x;
    }
    return res;
  }
}

File: Sheets/src/neetcode/practice/Math_And_Geometry/_6_PowXN.cc
class Solution {
 public:
  double myPow(double x, int n) {
    double res = 1;
    if (n < 0) {
      while (n++)
        res /= x;
    } else {
      while (n--)
        res *= x;
    }

    return res;
  }
};
